golang/pkg/builder/container/utils.go

```diff
@@ -176,6 +176,33 @@ func removeContainer(ctx context.Context, containerName string) error {
 	return nil
 }
 
+// Check the existence of a container, then return it, matches only one
+func getContainer(ctx context.Context, containerName string) (types.Container, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		panic(err)
+	}
+
+	filter := filters.NewArgs()
+	filter.Add("name", fmt.Sprintf("^%s$", containerName))
+
+	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{All: true, Filters: filter})
+	if err != nil {
+		return types.Container{}, err
+	}
+
+	switch len(containers) {
+	case 0:
+		// when we didn't matched any
+		return types.Container{}, nil
+	case 1:
+		return containers[0], nil
+	default:
+		// shouldn't happen
+		return types.Container{}, fmt.Errorf("invalid match")
```

Comment:
\* cannot happen, modify the error message to clearly communicate this, something like this: impossible, unreachable error, exact matching failed.

Reply:
last one sounds nice to me, thanks!